presto-main/src/main/java/com/facebook/presto/operator/repartition/OptimizedPartitionedOutputOperator.java

```diff
@@ -608,8 +607,9 @@ private static int getFixedWidthTypeSize(Type type)
 
         private int bufferedRowCount;
         private boolean bufferFull;
+        private OperatorContext operatorContext;
 
-        PartitionBuffer(int partition, int channelCount, int capacity, AtomicLong pagesAdded, AtomicLong rowsAdded, PagesSerde serde, Lifespan lifespan, ArrayAllocator bufferAllocator)
+        PartitionBuffer(int partition, int channelCount, int capacity, AtomicLong pagesAdded, AtomicLong rowsAdded, PagesSerde serde, Lifespan lifespan, ArrayAllocator bufferAllocator, OperatorContext operatorContext)
```

Comment:
Refactor the `PartitionBuffer` to remove the 7th parameter, since we're passing in `operatorContext`, and `lifeSpan` can be derived from `operatorContext` in the constructor.